src/coreclr/jit/codegencommon.cpp

```diff
@@ -3138,6 +3139,11 @@ void CodeGen::genSpillOrAddRegisterParam(unsigned lclNum, RegGraph* graph)
             graph->AddEdge(sourceReg, destReg, edgeType, seg.Offset - baseOffset);
         }
     }
+
+    if (isSpilled)
+    {
+        genHomeStackPartOfSplitParameter(lclNum);
+    }
```

Comment:
Can you move this handling outside `genHomeRegisterParams`? Prolog generations becomes much more constrained after `genHomeRegisterParams` has run, and this handling doesn't have anything to do with moving registers to their initial locations. I would also switch to use `initReg` instead of `REG_SCRATCH` for the same reason (using `REG_SCRATCH` wouldn't work on other platforms at the point this runs, for example. I guess it works because RISCV64 never allocates `REG_SCRATCH` for anything, but on other platforms that's not a valid register to clobber after `genHomeRegisterParams`.).

You can factor the code in the stack case from `genHomeSwiftStructParameters` to avoid duplicating that logic.

Reply:
My thinking was split parameter homing can happen can only if there are some argument registers to home (otherwise the parameter wouldn't be split), but ok, it can also be outside.